samples/ComputeSharp.SwapChain/Shaders/FourColorGradient.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace ComputeSharp.SwapChain.Shaders;
+
+/// <summary>
+/// A shader which interpolates four different colors into a rotating gradient pattern.
+/// Ported from <see href="https://www.shadertoy.com/view/7lc3R4"/>.
+/// <para>Created by Inigo Quilez.</para>
+/// <para>License Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License.</para>
+/// </summary>
+[AutoConstructor]
+internal readonly partial struct FourColorGradient : IPixelShader<float4>
+{
+    /// <summary>
+    /// The current time since the start of the application.
+    /// </summary>
+    public readonly float time;
+
+    // Colors to blend together
+    private static readonly float3 colorOne = new(0.999f, 0.999f, 0.999f);
+    private static readonly float3 colorTwo = new(0.999f, 0f, 0f);
+    private static readonly float3 colorThree = new(0f, 0.999f, 0f);
+    private static readonly float3 colorFour = new(0f, 0f, 0.999f);
+
+    /// <summary>
+    /// Standard 2D rotation formula.
+    /// </summary>
+    private static float2x2 Rotate(float a)
+    {
+        float c = Hlsl.Cos(a);
+        float s = Hlsl.Sin(a);
+
+        return new(c, s, -s, c);
+    }
+
+    /// <summary>
+    /// Generate a hash from a vector.
+    /// </summary>
+    /// <param name="p">Vector to hash</param>
```

Comment:
Nit: missing trailing '.' on XML parameters and returns.

Reply:
Fixed in all places in 58ed2478ed59c8a63794157b45b84239b9116c7f